enhancements/platform/quota-management/README.md

```diff
@@ -0,0 +1,1974 @@
+    the request the ability to proceed.
+    - **Mitigations (High-Level):**
+      - **Real-time Usage Queries:** The `quota-operator` should *always*
+        attempt to query the metering system (`amberflo`) in real-time for the
+        most up-to-date usage data before making a decision on a
+        `ResourceQuotaClaim`.
+      - **Cautious Use of Cached Data:** If the optional caching of usage in
+        `ResourceQuotaGrant.status.usage` is implemented, this data *must* be
+        treated as potentially stale. Decisions based purely on this cache
+        should be extremely limited (e.g., only for clear-cut denials where
+        requested > hard_limit_from_cache, never for granting claims if amberflo
+        is unreachable). Logic should *always* favor fresh data from the
+        metering engine.
+      - **Metering System Reliability:** The accuracy and timeliness of the
+        metering system itself are crucial and should be configured for high
+        availability and low data latency.
+      - **Clear Status Reporting:** If a claim is denied due to inability to
+        contact the metering system (despite retries), the
+        `ResourceQuotaClaim.status` should clearly reflect this reason (e.g.,
+        `MeteringSystemUnavailable`), distinguishing it from a denial due to
+        actual quota exhaustion.
+
+---
+
+## Design Details
+
+The Milo Quota Management System will be deployed as a series of components
+within the central Datum Management Control Plane (MCP). It will expose APIs and
+interact with Datum Cloud service instances, which are hosted within individual
+Project Control Planes (PCPs) for each tenant project.
+
+### Custom Resource Definitions
+
+Three main CRDs will be created as core components of the Quota Management
+implementation: `ServiceQuotaRegistration`, `ResourceQuotaClaim` and
+`ResourceQuotaGrant`. These CRDs will be defined and served by the **Milo
+APIServer**, a central component running within the Datum MCP. The API group for
+these CRDs will be `quota.miloapis.com`, aligning with existing API group
+standards; while the Go type definitions for these CRDs will be located within
+the [milo GitHub repository](https://github.com/datum-cloud/milo).
+
+Other `*.datumapis.com` API groups referenced in this document, such as
+`compute.datumapis.com` (hosting resources like `Instance`) and
+`network.datumapis.com` (hosting resources like `Subnet`), are served by
+individual **Project APIServers**. Each Project APIServer is dedicated to a
+specific tenant project and runs its own control plane. These Project APIServers
+are *not* where the Milo quota CRDs reside. However, they will be configured
+(via
+[multicluster-runtime](https://github.com/kubernetes-sigs/multicluster-runtime))
+to call Milo's admission webhooks (which run in the MCP and interact with the
+Milo APIServer) when tenant resources are created or modified.
```

Comment:
>  These Project APIServers are *not* where the Milo quota CRDs reside.

Something we need to decide is whether we're going to give users access to view the `ResourceQutoaClaims` for resources that exist within their project. If we do want users to be able to see these (I don't see why they shouldn't be able to), then we will need to register the CRD with the project control plane apiserver. 

> However, they will be configured (via [multicluster-runtime](https://github.com/kubernetes-sigs/multicluster-runtime)) to call Milo's admission webhooks (which run in the MCP and interact with the Milo APIServer) when tenant resources are created or modified.

This is going to be dependent on the integration strategy the service uses to manage quota resources with the quota system.